lightning/src/events/mod.rs

```diff
@@ -525,6 +525,31 @@ impl_writeable_tlv_based_enum_upgradable!(HTLCHandlingType,
 	},
 );
 
+/// The reason for HTLC failures in [`Event::HTLCHandlingFailed`].
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum HTLCHandlingFailureReason {
```

Comment:
This one does remind me of `HTLCFailReasonRepr`. Isn't there more in that struct that might be useful to expose here such as the failure data or the hold time?

Reply:
Definitely, this is essentially an external facing version of `HTLCFailReasonRepr`.

I don't think that we'll ever want to surface the encrypted data blob on the API, which is my main reason for not surfacing `HTLCFailReasonRepr`.

We're also not going to have tons and tons of fields that we need to translate from one to the other so I think it's nice to have a clean/readable api-facing struct.

Reply:
Definitely not the encrypted data blob. Perhaps there is something in the other fields/fail parameters that can be worth surfacing, but can be done when demand arises. There is this external facing version now.